Lesson6/hw_ig.py

```diff
@@ -0,0 +1,55 @@
+# 1. Implement a function that flatten incoming data:
+# non-iterables and elements from iterables (any nesting depth should be supported)
+# function should return an iterator (generator function)
+# don't use third-party libraries
+
+
+def merge_elems(*elems):
+    for iterable in elems:
+        if isinstance(iterable, (list, tuple)):
+            yield from merge_elems(*iterable)
+        elif not hasattr(iterable, '__iter__'):
+            yield iterable
+        else:
+            for item in iterable:
+                yield item
+
+
+# example input
+a = [1, 2, 3]
+b = 6
+c = 'zhaba'
+d = [[1, 2], [3, 4]]
+
+for _ in merge_elems(a, b, c, d):
+    print(_, end=' ')
+
+# output: 1 2 3 6 z h a b a 1 2 3 4
+
+# 2. Implement a map-like function that returns an iterator (generator function)
+# extra functionality: if arg function can't be applied, return element as is + text exception
+
+
+def map_like(func, *elems):
+    for i in elems:
+        try:
+            yield func(i)
+        except TypeError as e:
```

Comment:
There might be other exceptions apart from this one here (depends on the input)

Reply:
Fixed